docs/src/pages/components/data-grid/pagination/PageSizeAuto.js

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { DataGrid } from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+import Stack from '@mui/material/Stack';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import InputLabel from '@mui/material/InputLabel';
+import Select from '@mui/material/Select';
+
+export default function PageSizeAuto() {
+  const { data } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 100,
+    maxColumns: 6,
+  });
+
+  const [height, setHeight] = React.useState(400);
+
+  return (
+    <Stack style={{ width: '100%' }} alignItems="flex-start" spacing={2}>
+      <FormControl fullWidth>
+        <InputLabel
+          htmlFor="main-grouping-criteria"
+          id="ain-grouping-criteria-label"
+        >
+          Height of the container
+        </InputLabel>
+        <Select
+          label="Main Grouping Criteria"
+          onChange={(e) => setHeight(Number(e.target.value))}
+          value={height}
+          id="main-grouping-criteria"
+          labelId="main-grouping-criteria-label"
+        >
+          <MenuItem value="300">300px</MenuItem>
+          <MenuItem value="400">400px</MenuItem>
+          <MenuItem value="500">500px</MenuItem>
+        </Select>
+      </FormControl>
+      <div style={{ height, width: '100%' }}>
+        <DataGrid autoPageSize pagination {...data} />
```

Comment:
`pagination` prop is in each example, even if it is useless because they all use `DataGrid` and not `DataGridPro`. Is it to remind `DataGridPro` users they need this prop?

Reply:
Yes, it was @oliviertassinari proposal.
I think @m4theushw agree with me that we should drop it.

Reply:
Moreover, you insisted on the different default behavior between the two plans at the beginning. Could use a warning box to be sure devs do not miss it.

> The default pagination behavior depends on your plan.
> - On the `DataGrid`, pagination is enabled by default and can't be disabled
> - On the `DataGridPro`, pagination is disabled by default, use the `pagination` prop to enable it

Reply:
Done

Reply:
> Is it to remind DataGridPro users they need this prop?

It's to make sure that developers can replace `DataGrid` with `DataGridPro` without surprises.